request_a_govuk_domain/request/views.py

```diff
@@ -396,17 +401,37 @@ def send_confirmation_email(request) -> None:
 
 
 class SuccessView(View):
+    failure_url = reverse_lazy("service_failure")
+
     def get(self, request):
         reference = generate_reference()
-        save_data_in_database(reference, request)
-        if get_env_variable("SEND_EMAIL", "False") == "True":
-            send_confirmation_email(request)
+        try:
+            save_data_in_database(reference, request)
+
+            # NOTIFY_API_KEY is essential for Notify API to send email, which at the moment is being read from .env file
+            # Since all environment may not have .env file, and hence it will cause error if we try to send email,
+            # SEND_EMAIL has been added as a temporary flag, to not send email as default behaviour
+            # Once we get NOTIFY_API_KEY from AWS, then this flag should be removed
+            if get_env_variable("SEND_EMAIL", "False") == "True":
+                send_confirmation_email(request)
+        except Exception as e:
+            registration_data = request.session.get("registration_data", {})
+            logger.error(
+                f"""Exception at Application submitted stage. Exception: {type(e).__name__} - {str(e)} ,
+                         Registration data: {registration_data}"""
+            )
+            return HttpResponseRedirect(self.failure_url)
```

Comment:
I don't think this will return a 500 error page. It just redirects to a normal 200 page.